metadata_backend/api/auth.py

```diff
@@ -87,9 +88,8 @@ async def callback(self, req: Request) -> Response:
             LOG.error(reason)
             raise web.HTTPBadRequest(reason=reason)
 
-        # Verify, that states match
-        state = await self._get_from_session(req, "oidc_state")
-        if not secrets.compare_digest(str(state), str(params["state"])):
+        # Verify, that state is pending
+        if not params["state"] in req.app["OIDC_State"]:
```

Comment:
wondering why we don't need to compare digest here or why we dropped it ?

Reply:
With this PR, we delay session set up (cookie creation) to the callback, so we don't have the connection to compare the specific handed out `state` to the received. As a second best thing, we keep track of currently outstanding authentications (where callbacks have not yet been called).

I'm not sure I see understand what the security benefit of verifying the state is, but if one want to keep track of it, one can move the session creation (cookie setting) to `login`. The drawback of that (reason I did not do it for this PR) is that the cookie can not be completely filled in at that time and I've seen overall weird behaviour when changing cookies (cookies on redirects are also a thing, and I don't have a complete overview for what cases it works and when it's ignored).